src/sources/coinone.rs

```diff
@@ -89,6 +91,34 @@ pub struct Response {
     pub bid: Vec<PricePoint>,
 }
 
+///This trait returns a vector of ask prices and quantities
+impl AskBook for Response {
+    fn asks(&self) -> Result<Vec<PriceQuantity>, Error> {
+        let mut pq = vec![];
+        for p in self.ask.iter() {
+            pq.push(PriceQuantity {
+                price: p.price.clone(),
+                quantity: Decimal::from_str(&p.qty.clone())?,
+            });
+        }
+        return Ok(pq);
```

Comment:
Here's the fancy iterator based way (haven't tested) which I'm not sure actually makes it any more readable:

https://doc.rust-lang.org/stable/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect

```suggestion
        self.ask.iter().map(|p| {
            p.qty().parse().map(|quantity| PriceQuantity {
                price: p.price.clone(),
                quantity
            })
        }).collect()
```